rust_src/src/lisp.rs

```diff
@@ -155,6 +155,10 @@ impl LispObject {
     pub fn is_symbol(self) -> bool {
         self.get_type() == LispType::Lisp_Symbol
     }
+
+    pub fn symbol_name(&self) -> LispObject {
```

Comment:
This is not safe - it'll treat the `LispObject` as a symbol regardless of its type.

You'll have to introduce a `LispSymbolRef` newtype and respective `as_symbol` and `as_symbol_or_error` methods on `LispObject` to convert safely.